### PR TITLE
Bug 1770677: upstream the dnsrecord type

### PR DESCRIPTION
**Pull request by ironcladlou**
Depends on https://github.com/openshift/api/pull/584

### CODE REVIEW

**Comment by ironcladlou** [comment]
@Miciah @danehans @frobware PTAL

File: test/e2e/certificate_publisher_test.go [hunk 1]
```diff
@@ -9,8 +9,8 @@ import (
 
 	operatorv1 "github.com/openshift/api/operator/v1"
 
-	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
-	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
```
Comment (Miciah) [inline test/e2e/certificate_publisher_test.go]: Maybe import as `operatorcontroller`?  We are inconsistent in how we import this package, and `operatorcontroller` is the least ambiguous of the options in use.

Comment (ironcladlou) [reply test/e2e/certificate_publisher_test.go]: Agree, but not sure I agree with another round of CI for it...

**Pull request merged**